### PR TITLE
Support stripe payments 

### LINKED ISSUE #55: Integrate Stripe Checkout for donations
> This is a potential good first issue for someone in the community :)

I'd love for someone to integrate [client-only Checkout](https://stripe.com/docs/payments/checkout/client) basically replicating the functionality on https://engineers.sg/support_us ([see code](https://github.com/engineersftw/website/blob/master/app/views/welcome/support_us.html.slim)) with the different tiers which can be modeled with [plans](https://stripe.com/docs/payments/checkout/client-subscription#create-recurring-products) (for recurring donations) and [products/skus](https://stripe.com/docs/payments/checkout/client#create-products) (for one-time donations) and provide feedback about their experience integrating it.

I'm of course happy to provide guidance and support throughout if helpful. You can reach me via Twitter: https://twitter.com/thorwebdev

### PR DESCRIPTION
**Pull request by sarupbanskota**
Fixes #55 and also refactors the support template.

TODO:
- [x] Use Stripe instead of PayPal
- [x] Refactor submission to work for both modes
- [x] Clean up payment objects to make code readable
- [x] Failure/error states
- [x] Use actual keys / skus / plans

### CODE REVIEW

File: app/views/welcome/support_us.html.slim [hunk 1]
```diff
@@ -1,58 +1,90 @@
...
+      checkoutButtons[i].addEventListener('click', () => {
+        const chosenOption = document.getElementsByTagName('select')[i].value;
+        const customerEmail = document.getElementsByClassName('user_details')[i].value;
+
+        const productType = chosenOption.includes('plan_') ? 'plan' : 'sku';
+
+        const checkoutOptions = {
+          items: [{ [productType]: chosenOption, quantity: 1}],
+          successUrl: 'https://success.com',
+          cancelUrl: 'https://canceled.com'
```
Comment (sarupbanskota) [inline app/views/welcome/support_us.html.slim]: What do we wanna do here?

File: app/views/welcome/support_us.html.slim [hunk 2]
```diff
@@ -1,58 +1,90 @@
...
+        }
+
+        customerEmail ? checkoutOptions["customerEmail"] = customerEmail : null;
 
+        stripe.redirectToCheckout(checkoutOptions)
+        .then(function (result) {
+          if (result.error) {
```
Comment (sarupbanskota) [inline app/views/welcome/support_us.html.slim]: What do we wanna do here?

Comment (thorsten-stripe) [reply app/views/welcome/support_us.html.slim]: discussing options here: https://github.com/engineersftw/website/issues/55#issuecomment-571467676

File: app/views/welcome/support_us.html.slim (hunk 1 above)
Comment (thorsten-stripe) [reply app/views/welcome/support_us.html.slim]: discussing options here: https://github.com/engineersftw/website/issues/55#issuecomment-571467676

File: app/views/welcome/support_us.html.slim (hunk 2 above)
Comment (thorsten-stripe) [reply app/views/welcome/support_us.html.slim]: This is very rare (e.g. when connection drops etc.) Maybe we can just do an alert with the error message?

**Comment by thorsten-stripe** [comment]
@sarupbanskota qq, did you have issues running `bundle exec rake db:migrate`? I had to replace all instances of `ActiveRecord::Migration` with `ActiveRecord::Migration[4.2]` for the command to run.

**Comment by Drenmi** [comment]
@thorsten-stripe Sorry about that. I'm in the middle of upgrading the dependencies to modern versions. There will be some small hick-ups until I finish it up.

**Comment by Drenmi** [comment]
@sarupbanskota Is it beneficial in some way to remove PayPal as an option? I'm currently doing monthly donations through PayPal. What will happen to those subscriptions? 🤔

**Comment by sarupbanskota** [comment]
> @sarupbanskota qq, did you have issues running bundle exec rake db:migrate? I had to replace all instances of ActiveRecord::Migration with ActiveRecord::Migration[4.2] for the command to run.

Oh yeah, I forgot to mention that. That's actually what took me the longest in this exercise 😄 

My conclusion is that a new snapshot dump wasn't checked in after `db/migrate/20161113082712_add_slug_to_organizations_table.rb` was introduced, hence after you run the snapshot dump, Rails doesn't like that a migration is pending and suggests running a migration.

@Drenmi did I analyse right?

**Comment by sarupbanskota** [comment]
> @sarupbanskota Is it beneficial in some way to remove PayPal as an option? I'm currently doing monthly donations through PayPal. What will happen to those subscriptions? 🤔

[Disclaimer: I have nothing to with Stripe]

One benefit I can think of is that if you keep two options around — that's two places to track. Stripe's UX is arguably smoother, and it supports all the payment options PayPal does so I'd just keep one source. 

Existing recurring PayPal subscriptions shouldn't be affected because from what I gathered, there doesn't seem to be any controls on this website for the user to update them anyway. So if they'd like to cancel, my guess is that they do it through their PayPal account (same for Stripe).

**Comment by sarupbanskota** [comment]
@thorsten-stripe I've updated the keys, as well as added the failure state. 

Feel free to review (keys/plans from Stripe test mode).

**Comment by thorsten-stripe** [comment]
Thanks so much @sarupbanskota, works like a charm :)
I've added the Donate buttons mentioned above, I hope you don't mind:
`index.html`
image
`/support_us.html`
image


@miccheng would you able to quickly review and if you're 👍 can you swap in the live publishable key and the live sku and plan IDs, merge, and deploy? I've already applied the free processing volume to the Code Craft Stripe account, you'll only need to ack it from your Stripe Dashboard :)

**Comment by sarupbanskota** [comment]
@miccheng Lemme know if you need anything further from me, or if it's good to go!

**Comment by miccheng** [comment]
I'll try to clear this over the weekend. Sorry for the delay.

---
Regards,

Michael Cheng CM
Mobile: +65 9185 5166
LinkedIn: http://sg.linkedin.com/in/miccheng

**Comment by miccheng** [comment]
@thorsten-stripe @sarupbanskota I'll merge first and fix the values.

**Pull request merged**